### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by Norman209** [Norman209/Udacity-AI#1]

**Comment by Norman209** [comment]
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense, Flatten, Conv2D, MaxPooling2D,Dropout
import tensorflow as tf
import numpy as np
from tensorflow.keras.callbacks import LearningRateScheduler
import os
import cv2
counter = 0
def scheduler(epoch, lr):
    if epoch <= 2:
        return lr
    else:
        return lr * tf.math.exp(-0.1)
lr = 0.001
callback = LearningRateScheduler(scheduler)
a_folder = [i for i in os.listdir('/Users/milesnorman/autopilot_project/data_set/train_data') if i!='.DS_Store']
images = [i for i in a_folder if 'png' in i]
txt_files = [i for i in a_folder if 'txt' in i]
print('text files:',txt_files)
print('image files:',images)
x = []
y = []
nums = [str(i) for i in range(1000)]
txt_nums = []
for i in txt_files:
    print(i)
    image_num = [g for g in i if g in nums]
    image_num = int(''.join(image_num))
    print('txt num:', image_num)
    txt_nums.append(image_num)
    path = f'/Users/milesnorman/autopilot_project/data_set/train_data/{i}'
    with open(path,'r') as file:
        steering_angle = float(file.readline().split()[0])
        y.append(steering_angle)
    counter += 1
    print(f'{counter} txt file done')
counter = 0
for i in images:
    image_num = [g for g in i if g in nums]
    image_num = int(''.join(image_num))
    if image_num in txt_nums:
        print('image num:', image_num)
        path = f'/Users/milesnorman/autopilot_project/data_set/train_data/{i}'
        x.append(list(cv2.imread(path,0)))
        counter+=1
        print(f'{counter} image done')

x = np.array(x)
y = np.array(y)
print(len(y))
print(len(x))

def make_CNN(input_dim, hidden_neurons, n_actions):
    model = Sequential()
    model.add(Conv2D(input_shape=input_dim, filters=9, kernel_size=(6, 6), strides=(2, 2), padding='same',activation = 'relu'))#reduces
    model.add(MaxPooling2D(pool_size=(3, 3), strides=(3, 3)))
    model.add(Conv2D(filters=10, kernel_size=(8, 8), strides=(2, 2), padding='same',activation = 'relu'))
    model.add(MaxPooling2D(pool_size=(4, 4), strides=(4, 4)))
    model.add(Conv2D(filters=10, kernel_size=(8,8), strides=(2, 2), padding='same', activation='relu'))
    model.add(Conv2D(filters=10, kernel_size=(5, 5), strides=(1, 1), padding='same', activation='relu'))
    model.add(Flatten())
    model.add(Dense(units=hidden_neurons))
    model.add(Dropout(.3))
    model.add(Dense(units=hidden_neurons))
    model.add(Dense(units=hidden_neurons))
    model.add(Dense(units=n_actions))
    model.compile(optimizer='Adam', loss=tf.keras.losses.MSE)
    return model
batch_size = 50
epochs = len(y)//batch_size
model = make_CNN((1510,2938,1),40,1)
model.fit(x,y,batch_size = batch_size,epochs = epochs,callbacks = callback)
model.save('/Users/milesnorman/autopilot_project/self_driving_ai')
